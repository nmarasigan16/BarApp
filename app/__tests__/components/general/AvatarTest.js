import 'react-native';
import React from 'react';
import Avatar from '../../../src/components/general/avatar/AvatarComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <Avatar
            image={'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png'}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

describe('renders sizes', () => {
    it('renders small', () => {
        const tree = renderer.create(
            <Avatar
                image={'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png'}
                size={'small'}/>
        ).toJSON();
    });

    it('renders medium', () => {
        const tree = renderer.create(
            <Avatar
                image={'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png'}
                size={'medium'}/>
        ).toJSON();
    });

    it('renders large', () => {
        const tree = renderer.create(
            <Avatar
                image={'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png'}
                size={'large'}/>
        ).toJSON();
    });

    it('renders xlarge', () => {
        const tree = renderer.create(
            <Avatar
                image={'https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png'}
                size={'xlarge'}/>
        ).toJSON();
    });

});