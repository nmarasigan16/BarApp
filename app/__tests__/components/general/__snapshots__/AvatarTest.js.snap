// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://assets-cdn.github.com/images/modules/logos_page/GitHub-Mark.png",
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 17,
        "height": 34,
        "width": 34,
      },
      undefined,
    ]
  }
/>
`;
